### ISSUE TITLE
Resolve node queries directly from store

### ISSUE THREAD

**Issue by stubailo** [apollostack/apollo-client#79]
When we have a query like:

```
{
  node(id: "1") {
    __typename,
    id,
    ... on Person {
      name
    }
  }
}
```

If we already have a node with ID 1 in the store, we should be able to retrieve it. Right now, this will not realize that this is just looking for a particular node, and hit the network with a new root query. I guess this also means, when we receive the result of a root node query, we might want to not cache it under the root query, but instead only under the ID of the node.

**Comment by stubailo** [comment]
This will become important once fragment variables are a thing. But isn't as useful now.

**Comment by stubailo** [comment]
Could be implemented using #26
